README.md
reactBigCalendarAdapter | [Дизайн-система Consta](https://consta.design/)

Этот адаптер стилизует таблицу [react-big-calendar](http://jquense.github.io/react-big-calendar/examples/index.html?path=/story/about-big-calendar--page) для [дизайн-системы Consta](https://consta.design/).

# Как использовать

## Установите пакет

```sh
# NPM
$ npm install @consta/react-big-calendar-adapter
# Yarn
$ yarn add @consta/react-big-calendar-adapter
```

## Подключите зависимости

Чтобы начать работу, установите библиотеку [`@consta/uikit`](https://www.npmjs.com/package/@consta/uikit) и [настройте тему](https://consta.design/libs/portal/theme-themeabout).

### Можно использовать компоненты

Например, так:

```js
import React from 'react';
import { reactBigCalendarAdapter } from '@consta/react-big-calendar/reactBigCalendarAdapter';
import moment from 'moment';
import { momentLocalizer, Calendar } from 'react-big-calendar';
moment.locale('ru');
const localizer = momentLocalizer(moment);
export const ReactBigCalendarAdapter = () => {
  const { prefix, ...otherProps } = reactBigCalendarAdapter({});
  return (
    <Calendar
      defaultDate={new Date(2022, 3, 1)}
      events={events}
      localizer={localizer}
      showMultiDayTimes
      className={prefix}
      defaultView="month"
      {...otherProps}
    />
  );
};
```

## Документация

[Посмотреть документацию и примеры](http://consta.design/libs/react-big-calendar-adapter)

## Разработка

### Подготовка окружения

Рабочее окружение должно содержать NodeJS и Yarn.

Чтобы установить зависимости, выполните команду:

```sh
$ yarn install
```

### Основные команды

```sh
# Запуск локального сервера для разработки
$ yarn start

# Сборка пакета
$ yarn build

# Сборка стенда
$ yarn stand:build

# Запуск тестов
$ yarn test
```

## Контрибьюторам

Будем рады, если вы захотите принять участие в разработке дизайн-системы =) Но сначала прочитайте [инструкцию для контрибьюторов](https://consta.design/libs/portal/contributers-code).

## Лицензия

Дизайн-систему можно использовать бесплатно, она распространяется на условиях открытой [лицензии MIT](https://consta.design/static/licence_mit.pdf).


CHANGELOG.md
# Changelog

## v0.2.3 (23/05/2023)
Самое важное:
- Запускать сервер разработки и собирать пакеты на ОС Windows теперь можно во всех библиотеках Consta.
- Добавили год в заголовок календаря при квартальном отображении.

---

- [chore: configure build for windows (#17)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/5b5efc02ebb1efe454caf937236eca4ac97c85a2) - [@gizeasy](https://github.com/gizeasy)
- [feat(QuartalView): add year to title (#16)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/d6b9b5903ffe543c6a3fb2ec0412e5031ad477d5) - [@N1MBER](https://github.com/N1MBER)
- [docs(ReactBigCalendarAdapter): add aliases (#15)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/a7a2eb24bde23ad902bccafab7febfc182c383f5) - [@N1MBER](https://github.com/N1MBER)
- [chore: update @consta/stand](https://github.com/consta-design-system/react-big-calendar-adapter/commit/cf9c65223c5978e22ef88867f9a8c2548e23f00f) - [@gizeasy](https://github.com/gizeasy)

--------------------

## v0.2.2 (30/11/2022)
- [fix(prefix): add prefix for class names (#14)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/065e14412ac355ce5daa381fcffd6fd5f6fd539e) - [@N1MBER](https://github.com/N1MBER)

--------------------

## v0.2.1 (24/11/2022)
- [fix(reactBigCalendarAdapter): fix z-index for overlay (#13)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/f9b734cbf71941544fddd370a50f9c0498ae7bbe) - [@N1MBER](https://github.com/N1MBER)
- [docs(portal): readme, version, stand (#12)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/bd239602624c88979996538bf52f923b56537c75) - [@arhayka](https://github.com/arhayka)
- [docs: readme, menu, links, start (#11)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/8184305123faecf667d7254b75792fc089e5819c) - [@arhayka](https://github.com/arhayka)
- [chore(deps): update @consta/stand](https://github.com/consta-design-system/react-big-calendar-adapter/commit/25b4f0b7fd93105791ff578baada49eccbbac260) - [@gizeasy](https://github.com/gizeasy)

--------------------

## v0.2.0 (29/09/2022)
Самое важное:
Выкладываем новый стенд и сборщик.

Добавили песочницу: теперь в ней можно смотреть примеры компонентов.

---

- [chore(build): new builder and new stand (#5)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/89bd900f250cf9cd2e80f0ef42efabec67cc80d0) - [@gizeasy](https://github.com/gizeasy)

--------------------

## v0.1.1 (10/08/2022)
- [refactor(Calendar): chnage month names and quartla view (#4)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/a2dc3f7c1e1343d7a300b5115a9d09485346a06b) - [@N1MBER](https://github.com/N1MBER)
- [chore(gh-actions): add actions (#2)](https://github.com/consta-design-system/react-big-calendar-adapter/commit/57bcce2c3ce5cc796b54a061f78b22ea93a1d128) - [@gizeasy](https://github.com/gizeasy)

--------------------

## v0.1.0 (08/07/2022)
Сделали адаптер к календарю [react-big-calendar](https://jquense.github.io/react-big-calendar/examples/?path=/docs/about-big-calendar--page): с его помощью этот календарь можно использовать вместе с другими компонентами дизайн-системы `Consta`.

---

- [chore(package.json): change name of package](https://github.com/consta-design-system/react-big-calendar-adapter/commit/733ebcc1a3b2065e5e25d08efd2dec405daafec0) - [@N1MBER](https://github.com/N1MBER)
- [feat(reactCalendarAdapter): add new adapter](https://github.com/consta-design-system/react-big-calendar-adapter/commit/0347d9e60388495e1e30a37ce9c2cfd25e9f924b) - [@N1MBER](https://github.com/N1MBER)
- [feat(Adapter): change text and fix bugs](https://github.com/consta-design-system/react-big-calendar-adapter/commit/38b1ed16a6b003b9cffab50146ae2e694bf7fe18) - [@N1MBER](https://github.com/N1MBER)
- [docs(readme): add readme](https://github.com/consta-design-system/react-big-calendar-adapter/commit/074c155eb0048d2d0b3e37470a57967176f2e87f) - [@arhayka](https://github.com/arhayka)
- [docs(reactBigCalendarAdapter): edit docs](https://github.com/consta-design-system/react-big-calendar-adapter/commit/d88e094a84a368feb300378eb53f77660c66d001) - [@arhayka](https://github.com/arhayka)
- [feat(reactCalendarAdapter): add new adapter](https://github.com/consta-design-system/react-big-calendar-adapter/commit/8482d271b8b67e31619298bb9a499b027d7e3113) - [@N1MBER](https://github.com/N1MBER)
- [chore(Calendar): add main files for lib](https://github.com/consta-design-system/react-big-calendar-adapter/commit/727da699bb3eb3c826fd0ebfcbd0cabe081f7683) - [@N1MBER](https://github.com/N1MBER)
- [Initial commit](https://github.com/consta-design-system/react-big-calendar-adapter/commit/5ef362899f8ee5e1213e5f0dd00469e90d853b42) - [@gizeasy](https://github.com/gizeasy)


src/adapters/reactBigCalendarAdapter/__mocks__/mock.data.ts
const now = new Date();

export const events = [
  {
    id: 0,
    title: 'Собыите на весь день с очень длинным названием',
    allDay: true,
    start: new Date(2022, 3, 0),
    end: new Date(2022, 3, 1),
  },
  {
    id: 1,
    title: 'Долгое событие',
    start: new Date(2022, 3, 7),
    end: new Date(2022, 3, 10),
  },

  {
    id: 2,
    title: 'Что-то начинается',
    start: new Date(2016, 2, 13, 0, 0, 0),
    end: new Date(2016, 2, 20, 0, 0, 0),
  },

  {
    id: 3,
    title: 'Что-то заканчивается',
    start: new Date(2016, 10, 6, 0, 0, 0),
    end: new Date(2016, 10, 13, 0, 0, 0),
  },

  {
    id: 4,
    title: 'Просто событие',
    start: new Date(2022, 3, 9, 0, 0, 0),
    end: new Date(2022, 3, 10, 0, 0, 0),
  },
  {
    id: 5,
    title: 'Конференция',
    start: new Date(2022, 3, 11),
    end: new Date(2022, 3, 13),
    desc: 'Важная конференция для важных людей',
  },
  {
    id: 6,
    title: 'Встреча',
    start: new Date(2022, 3, 12, 10, 30, 0, 0),
    end: new Date(2022, 3, 12, 12, 30, 0, 0),
    desc: 'Встреча для подготовки к другой встрече',
  },
  {
    id: 7,
    title: 'Обед',
    start: new Date(2022, 3, 12, 12, 0, 0, 0),
    end: new Date(2022, 3, 12, 13, 0, 0, 0),
    desc: 'Звездная призма, дай мне сил',
  },
  {
    id: 8,
    title: 'Встреча',
    start: new Date(2022, 3, 12, 14, 0, 0, 0),
    end: new Date(2022, 3, 12, 15, 0, 0, 0),
  },
  {
    id: 9,
    title: 'Нельзя пропустить',
    start: new Date(2022, 3, 12, 17, 0, 0, 0),
    end: new Date(2022, 3, 12, 17, 30, 0, 0),
    desc: 'Второй обед или завтрак',
  },
  {
    id: 10,
    title: 'Ужин',
    start: new Date(2022, 3, 12, 20, 0, 0, 0),
    end: new Date(2022, 3, 12, 21, 0, 0, 0),
  },
  {
    id: 11,
    title: 'Планируем встречу с боссом',
    start: new Date(2022, 3, 13, 8, 0, 0),
    end: new Date(2022, 3, 13, 10, 30, 0),
  },
  {
    id: 11.1,
    title: 'Крайне неудобный созвон',
    start: new Date(2022, 3, 13, 9, 30, 0),
    end: new Date(2022, 3, 13, 12, 0, 0),
  },
  {
    id: 11.2,
    title: 'Закрываем надоевший проект',
    start: new Date(2022, 3, 13, 11, 30, 0),
    end: new Date(2022, 3, 13, 14, 0, 0),
  },
  {
    id: 11.3,
    title: 'Ретро',
    start: new Date(2022, 3, 13, 15, 30, 0),
    end: new Date(2022, 3, 13, 16, 0, 0),
  },
  {
    id: 12,
    title: 'Что-то поздно вечером',
    start: new Date(2022, 3, 17, 19, 30, 0),
    end: new Date(2022, 3, 18, 2, 0, 0),
  },
  {
    id: 12.5,
    title: 'Что-то еще позже',
    start: new Date(2022, 3, 17, 19, 30, 0),
    end: new Date(2022, 3, 17, 23, 30, 0),
  },
  {
    id: 13,
    title: 'Событие на несколько дней',
    start: new Date(2022, 3, 20, 19, 30, 0),
    end: new Date(2022, 3, 22, 2, 0, 0),
  },
  {
    id: 14,
    title: 'Сегодня',
    start: new Date(new Date().setHours(new Date().getHours() - 3)),
    end: new Date(new Date().setHours(new Date().getHours() + 3)),
  },
  {
    id: 15,
    title: 'Вы находитесь здесь',
    start: now,
    end: now,
  },
  {
    id: 16,
    title: 'Записываем видео',
    start: new Date(2022, 3, 14, 15, 30, 0),
    end: new Date(2022, 3, 14, 19, 0, 0),
  },
  {
    id: 17,
    title: 'Записываем «В лесу родилась елочка»',
    start: new Date(2022, 3, 14, 16, 30, 0),
    end: new Date(2022, 3, 14, 20, 0, 0),
  },
  {
    id: 18,
    title: 'Хэллоуин',
    start: new Date(2022, 3, 14, 16, 30, 0),
    end: new Date(2022, 3, 14, 17, 30, 0),
  },
  {
    id: 19,
    title: 'Онлайн-собеседование',
    start: new Date(2022, 3, 14, 17, 30, 0),
    end: new Date(2022, 3, 14, 20, 30, 0),
  },
  {
    id: 20,
    title: 'Накладывающееся собыите',
    start: new Date(2022, 3, 14, 17, 0, 0),
    end: new Date(2022, 3, 14, 18, 30, 0),
  },
  {
    id: 21,
    title: 'Собеседование по телефону',
    start: new Date(2022, 3, 14, 17, 0, 0),
    end: new Date(2022, 3, 14, 18, 30, 0),
  },
  {
    id: 22,
    title: 'Регата',
    start: new Date(2022, 3, 14, 17, 30, 0),
    end: new Date(2022, 3, 14, 19, 0, 0),
  },
  {
    id: 23,
    title: 'В спорзал',
    start: new Date(2022, 3, 14, 18, 30, 0),
    end: new Date(2022, 3, 14, 20, 0, 0),
  },

  {
    id: 24,
    title: 'Онлайн-собеседование',
    start: new Date(2022, 6, 14, 17, 30, 0),
    end: new Date(2022, 6, 14, 20, 30, 0),
  },
  {
    id: 25,
    title: 'Накладывающееся событие',
    start: new Date(2022, 6, 14, 17, 0, 0),
    end: new Date(2022, 6, 14, 18, 30, 0),
  },
  {
    id: 26,
    title: 'Собеседование по телефону',
    start: new Date(2022, 7, 14, 17, 0, 0),
    end: new Date(2022, 7, 14, 18, 30, 0),
  },
  {
    id: 27,
    title: 'Регата',
    start: new Date(2022, 7, 14, 17, 30, 0),
    end: new Date(2022, 7, 14, 19, 0, 0),
  },
  {
    id: 28,
    title: 'В спорзал',
    start: new Date(2022, 5, 14, 18, 30, 0),
    end: new Date(2022, 5, 14, 20, 0, 0),
  },
];

export const backgroundEvent = [
  {
    id: 0,
    title: 'Только для пользователей',
    start: new Date(2022, 3, 13, 6),
    end: new Date(2022, 3, 13, 18),
  },
];
